File: src/ocaml/server/translator/location_history.ml
module D = Sxfiler_domain.Location_history
module G = Sxfiler_server_generated

let of_domain (t : D.t) =
  {
    G.Filer.LocationHistory.records = List.map Location_record.of_domain t.D.records;
    maxRecordNumber = t.max_record_num;
  }

let to_domain (t : G.Filer.LocationHistory.t) =
  { D.records = List.map Location_record.to_domain t.records; max_record_num = t.maxRecordNumber }

File: src/ts/domains/location-history.ts
// histories of location moved filer before

export type LocationRecord = {
  readonly location: string;
  readonly timestamp: Date;
};

export type LocationHistory = {
  readonly records: LocationRecord[];
  readonly maxRecordNumber: number;
};

// creare new LocationHistory
export const createLocationHistory = function createLocationHistory({
  records,
  maxRecordNumber,
}: LocationHistory): LocationHistory {
  return {
    records,
    maxRecordNumber,
  };
};
